src/core/expression/qgsexpressioncontextutils.cpp

```diff
@@ -801,18 +802,25 @@ QVariant QgsExpressionContextUtils::GetLayerVisibility::func( const QVariantList
     return false;
   }
 
+  bool isVisible = false;
   QgsMapLayer *layer = QgsExpressionUtils::getMapLayer( values.at( 0 ), parent );
-  if ( layer )
+  if ( layer && mLayers.contains( layer ) )
```

Comment:
This isn't thread safe -- you'll need to record the min/max scales for each layer in the constructor and use them here

Reply:
Are you saying this function was _never_ thread safe until now?